src/python/pants/engine/scheduler.py

```diff
@@ -406,15 +406,25 @@ def visualize_graph_to_file(self, filename):
   def visualize_rule_graph_to_file(self, filename):
     self._scheduler.visualize_rule_graph_to_file(filename)
 
+  def execution_request_literal(self, request_specs):
+    native_execution_request = self._scheduler._native.new_execution_request()
+    for subject, product in request_specs:
+      self._scheduler.add_root_selection(native_execution_request, subject, product)
+    return ExecutionRequest(request_specs, native_execution_request)
+
+  # FIXME: note that this does a cross prod!
   def execution_request(self, products, subjects):
     """Create and return an ExecutionRequest for the given products and subjects.
 
-    The resulting ExecutionRequest object will contain keys tied to this scheduler's product Graph, and
-    so it will not be directly usable with other scheduler instances without being re-created.
+    The resulting ExecutionRequest object will contain keys tied to this scheduler's product Graph,
+    and so it will not be directly usable with other scheduler instances without being re-created.
 
-    An ExecutionRequest for an Address represents exactly one product output, as does SingleAddress. But
-    we differentiate between them here in order to normalize the output for all Spec objects
-    as "list of product".
+    An ExecutionRequest for an Address represents exactly one product output, as does
+    SingleAddress. But we differentiate between them here in order to normalize the output for all
+    Spec objects as "list of product".
```

Comment:
I don't think this is true any longer... can nuke this paragraph probably.

Reply:
Done in 5a1665b!